From 362f86369d1930ad177acaa47225d24d26b02c8d Mon Sep 17 00:00:00 2001
From: Colin Ian King <colin.i.king@intel.com>
Date: Tue, 14 Nov 2023 13:29:45 +0000
Subject: [PATCH 171/180] sched/fair: remove upper limit on cpu number

Remove the 8-CPU cap on get_update_sysctl_factor(). On systems with
more than 8 cores (e.g. Zen 4 with 16+ threads), the scheduler tuning
parameters (sched_migration_cost, sched_nr_migrate) should scale with
the actual number of online CPUs rather than being capped.

Signed-off-by: Colin Ian King <colin.i.king@intel.com>
---
 kernel/sched/fair.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/sched/fair.c b/kernel/sched/fair.c
index 2048138ce54b..903ead0afacb 100644
--- a/kernel/sched/fair.c
+++ b/kernel/sched/fair.c
@@ -194,7 +194,7 @@ static inline void update_load_set(struct load_weight *lw, unsigned long w)
  */
 static unsigned int get_update_sysctl_factor(void)
 {
-	unsigned int cpus = min_t(unsigned int, num_online_cpus(), 8);
+	unsigned int cpus = num_online_cpus();
 	unsigned int factor;

 	switch (sysctl_sched_tunable_scaling) {
--
2.47.1

